element-{web,desktop}: 1.7.31 -> 1.7.33

Ma27 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

:warning: **Note:** please note that I'm not an active `element-desktop`-user. While it seems to work fine, it would be good if somebody could test it.

`element-web`:
* https://github.com/vector-im/element-web/releases/tag/v1.7.33
* https://github.com/vector-im/element-web/releases/tag/v1.7.33-rc.1
* https://github.com/vector-im/element-web/releases/tag/v1.7.32
* https://github.com/vector-im/element-web/releases/tag/v1.7.32-rc.1

`element-desktop`:
* https://github.com/vector-im/element-desktop/releases/tag/v1.7.33
* https://github.com/vector-im/element-desktop/releases/tag/v1.7.32

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Ma27 committed:
element-web: 1.7.31 -> 1.7.33

ChangeLogs:
* https://github.com/vector-im/element-web/releases/tag/v1.7.33
* https://github.com/vector-im/element-web/releases/tag/v1.7.33-rc.1
* https://github.com/vector-im/element-web/releases/tag/v1.7.32
* https://github.com/vector-im/element-web/releases/tag/v1.7.32-rc.1

Ma27 committed:
element-desktop: 1.7.31 -> 1.7.33

ChangeLogs:
* https://github.com/vector-im/element-desktop/releases/tag/v1.7.33
* https://github.com/vector-im/element-desktop/releases/tag/v1.7.32

sumnerevans commented:
I get the following error:

```
Changing application language to en-us
Fetching translation json for locale: en-us
Could not fetch translation json for locale: 'en-us' Error: Cannot find module './i18n/strings/en-us.json'
Require stack:
- /nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/language-helper.js
- /nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/tray.js
- /nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/electron-main.js
- /nix/store/kjv95jyji0a6587l7ljnnpb56vlb2iha-electron-13.1.6/lib/electron/resources/default_app.asar/main.js
-
    at Module._resolveFilename (internal/modules/cjs/loader.js:887:15)
    at Function.n._resolveFilename (electron/js2c/browser_init.js:257:1128)
    at Module._load (internal/modules/cjs/loader.js:732:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
    at Module.require (internal/modules/cjs/loader.js:959:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at AppLocalization.fetchTranslationJson (/nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/language-helper.js:76:20)
    at /nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/language-helper.js:89:39
    at Array.forEach (<anonymous>)
    at AppLocalization.setAppLocale (/nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/language-helper.js:88:17) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/language-helper.js',
    '/nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/tray.js',
    '/nix/store/rfsgqqv6hxrkq7k9zqxfg2v5gnwmi4ir-element-desktop-1.7.33/share/element/electron/lib/electron-main.js',
    '/nix/store/kjv95jyji0a6587l7ljnnpb56vlb2iha-electron-13.1.6/lib/electron/resources/default_app.asar/main.js',
    undefined
  ]
}
```
which causes
![image](https://user-images.githubusercontent.com/16734772/126814103-3fb1971f-c1d9-4b8d-9caa-7a9d6fb9c435.png)

Ma27 committed:
element-desktop: try to fix localization issues

I know that the en_EN.json vs. en-us.json is a nasty hack, but I don't
really understand where this `en-us.json` is supposed to be coming from.

sumnerevans commented:
Your latest commit fixed the translation error.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 131217` at b856b195 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>element-desktop</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:334:7:
```
    |
334 |       installPhase = attrs.installPhase or ''
    |       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

rsync is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:294:7:
```
    |
294 |       buildInputs = [ yarn nodejs rsync ] ++ extraBuildInputs;
    |       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

yarn is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:294:7:
```
    |
294 |       buildInputs = [ yarn nodejs rsync ] ++ extraBuildInputs;
    |       ^
```
</li>
  </ul>
</details>

blitz commented:
I'm an active element-desktop user and I haven't noticed any issues. :tada:

blitz commented on pkgs/applications/networking/instant-messengers/element/element-desktop.nix:
```diff
@@ -24,6 +24,17 @@ in mkYarnPackage rec {
 
   nativeBuildInputs = [ makeWrapper ];
 
+  buildPhase = ''
```
For posterity, it would be helpful to have a comment here why this is needed.
